sdk/storage/azure_storage_blob/src/clients/blob_client.rs

```diff
@@ -149,6 +149,19 @@ impl BlobClient {
         Ok(response)
     }
 
+    /// Marks the specified blob for deletion. The blob is later deleted during garbage collection.
```

Comment:
This documentation string has much confusion in the past and should not be used. Please just change to something like "Deletes the Blob". We should eventually update the tsp to reflect this.